spinn_front_end_common/interface/interface_functions/dsg_region_reloader.py

```diff
@@ -29,96 +28,149 @@
 from spinn_front_end_common.utilities.helpful_functions import (
     generate_unique_folder_name)
+from spinn_front_end_common.utilities.constants import BYTES_PER_WORD
 
 
-def dsg_region_reloader():
-    reloader = _DSGRegionReloader()
-    # pylint: disable=protected-access
-    reloader._run()
+def reload_dsg_regions():
+    """ Reloads DSG regions where needed
+    """
+    # build file paths for reloaded stuff
+    data_dir = get_reload_data_dir()
+
+    progress = ProgressBar(
+        FecDataView.get_n_placements(), "Reloading data")
+    for placement in progress.over(FecDataView.iterate_placemements()):
+        # Generate the data spec for the placement if needed
+        regenerate_data_spec(placement, data_dir)
+
+    # App data directory can be removed as should be empty
+    os.rmdir(data_dir)
 
 
-class _DSGRegionReloader(object):
-    """ Regenerates and reloads the data specifications.
+def get_reload_data_dir():
+    """ Get a path in which reloaded data files can be written
+
+    :rtype: str
+    """
+    run_dir_path = FecDataView.get_run_dir_path()
+    data_dir = generate_unique_folder_name(
+        run_dir_path, "reloaded_data_regions", "")
+    if not os.path.exists(data_dir):
+        os.makedirs(data_dir)
+    return data_dir
+
+
+def regenerate_data_spec(placement, data_dir):
+    """ Regenerate a data specification for a placement
+
+    :param ~.Placement placement: The placement to regenerate
+    :param str data_dir: A place to use to write data to
+    :return: Whether the data was regenerated or not
+    :rtype: bool
+    """
+    vertex = placement.vertex
+
+    # If the vertex doesn't regenerate, skip
+    if not isinstance(vertex, AbstractRewritesDataSpecification):
+        return False
+
+    # If the vertex doesn't require regeneration, skip
+    if not vertex.reload_required():
+        return False
+
+    txrx = FecDataView.get_transceiver()
+
+    # build the writers for the reports and data
+    spec_file, spec = get_data_spec_and_file_writer_filename(
+        placement.x, placement.y, placement.p, data_dir)
+
+    # Execute the regeneration
+    vertex.regenerate_data_specification(spec, placement)
+
+    # execute the spec
+    with open(spec_file, "rb") as spec_reader:
+        data_spec_executor = DataSpecificationExecutor(
+            spec_reader, SDRAM.max_sdram_found)
+        data_spec_executor.execute()
+    try:
+        os.remove(spec_file)
+    except Exception:  # pylint: disable=broad-except
+        # Ignore the deletion of files as non-critical
+        pass
+
+    # Read the region table for the placement
+    regions_base_address = txrx.get_cpu_information_from_core(
+        placement.x, placement.y, placement.p).user[0]
+    start_region = get_region_base_address_offset(regions_base_address, 0)
+    table_size = get_region_base_address_offset(
+        regions_base_address, MAX_MEM_REGIONS) - start_region
+    ptr_table = numpy.frombuffer(txrx.read_memory(
```

Comment:
You set a value and the overwrite that value in the next line!

Reply:
True, but the function takes the ptr_table generated in this line in...

Reply:
Stand alone it would then be cleaner if you move the create ptr_table into the second function.

If however you intend to use the second function elsewhere than fine.

Reply:
It is only this function that is visible to the outside, so I have moved the code for getting the pointer table into the function for fixing it (and renamed it of course)